raft/log_test.go

```diff
@@ -281,9 +281,7 @@ func TestLog_WriteEntriesTo_Snapshot(t *testing.T) {
 	}()
 
 	// Wait for buffer to be written to.
-	for buf.Len() == 0 {
-		runtime.Gosched()
-	}
+	time.Sleep(1 * time.Second)
```

Comment:
I'd need to check the code, but I thought we put a lock into `Len`, such that access to the underlying buffer was atomic (I need to check). And you may have found a different issue too.